modules/candidate_parameters/jsx/candidateInfo.js

```diff
@@ -0,0 +1,324 @@
+                    {alertMessage}
+                </div>
+                <FormElement
+                    name ="candidateInfo"
+                    onSubmit ={this.handleSubmit}
+                    ref ="form"
+                    class ="col-md-6"
+                >
+                    <StaticElement
+                        label ="PSCID"
+                        text ={this.state.Data.pscid}
+                    />
+                    <StaticElement
+                        label ="DCCID"
+                        text ={this.state.Data.candID}
+                    />
+                    <SelectElement
+                        label ="Caveat Emptor Flag for Candidate"
+                        name ="flagged_caveatemptor"
+                        options ={this.state.caveatOptions}
+                        value ={this.state.Data.flagged_caveatemptor}
+                        onUserInput ={this.setFormData}
+                        ref ="flagged_caveatemptor"
+                        disabled ={disabled}
+                        required ={true}
+                    />
+                    <SelectElement
+                        label ="Reason for Caveat Emptor Flag"
+                        name ="flagged_reason"
+                        options ={this.state.Data.caveatReasonOptions}
+                        value ={this.state.Data.flagged_reason}
+                        onUserInput ={this.setFormData}
+                        ref ="flagged_reason"
+                        disabled ={reasonDisabled}
+                        required ={!reasonDisabled}
+                    />
+                    {otherText}
+                    {extraParameterFields}
+                    {updateButton}
+                </FormElement>
+            </div>
+            );
+    },
+        /**
+     * Handles form submission
+     *
+     * @param {event} e - Form submission event
+     */
+    handleSubmit: function(e) {
+      e.preventDefault();
+      var myFormData = this.state.formData;
+            // Set form data and upload the media file
+      var self = this;
+      var formData = new FormData();
+      for (var key in myFormData) {
+        if (myFormData.hasOwnProperty(key)) {
+          if (myFormData[key] !== "") {
+            formData.append(key, myFormData[key]);
+          }
+        }
+      }
+
+      formData.append('tab', this.props.tabName);
+      formData.append('candID', this.state.Data.candID);
+      $.ajax(
+        {
+          type: 'POST',
+          url: self.props.action,
+          data: formData,
+          cache: false,
+          contentType: false,
+          processData: false,
+          success: function(data) {
+            self.setState(
+              {
+                updateResult: "success"
+              }
+                        );
+          },
+          error: function(err) {
+            var errorMessage = JSON.parse(err.responseText).message;
```

Comment:
If a random error occurs on the server, `err.responseText` is set to `""` and this line generates an error.
So you should probably check for empty and/or fallback on statusText